pkg/helm/controller.go

```diff
@@ -150,21 +164,46 @@ func (c *Controller) OnHelmRemove(key string, chart *helmv1.HelmChart) (*helmv1.
 	return newChart, c.apply.WithOwner(newChart).Apply(objectset.NewObjectSet())
 }
 
+func (c *Controller) OnConfChange(key string, conf *helmv1.HelmChartConfig) (*helmv1.HelmChartConfig, error) {
```

Comment:
missing a check for `conf == nil` which is necessary because sometimes when deleting a CR you will get OnChange events with a non-empty key but a nil CR ref.